skills-kit-library/test/skills-kit-2.0.test.js

```diff
@@ -0,0 +1,512 @@
+            status: { state: 'pending' },
+            content: { url_template: 'url_template' },
+            info: { url: 'https://www.box.com' }
+        };
+        const reps = { entries: [repInfo] };
+        const fileGetResponse = {
+            statusCode: 200,
+            body: {
+                status: {
+                    state: 'success'
+                },
+                content: {
+                    url_template: 'polled_url_template'
+                }
+            }
+        };
+
+        reader.fileReadClient = {
+            files: {
+                getRepresentationInfo: jest.fn().mockReturnValue(Promise.resolve(reps))
+            },
+            get: jest.fn().mockReturnValue(Promise.resolve(fileGetResponse))
+        };
+
+        return reader.getBasicFormatFileURL().then((data) => {
+            expect(data).toEqual('polled_url_template?access_token=readtoken12345');
+        });
+    });
+
+    test('getBasicFormatFileURL() should throw exception for unknown representation status', () => {
+        const repInfo = { status: { state: 'error' } };
+        const reps = { entries: [repInfo] };
+        console.error = jest.fn();
+
+        reader.fileReadClient = {
+            files: {
+                getRepresentationInfo: jest.fn().mockReturnValue(Promise.resolve(reps))
+            }
+        };
+
+        return reader.getBasicFormatFileURL().catch((error) => {
+            expect(error.message).toEqual(SkillsErrorEnum.FILE_PROCESSING_ERROR);
+            expect(console.error).toHaveBeenCalledWith('Representation had error status');
+        });
+    });
+});
+
+describe('SkillsWriter', () => {
+    let writer;
+    const mockDate = new Date().toISOString();
+    const dt = {
+        toISOString: jest.fn().mockReturnValue(mockDate)
+    };
+    global.Date = jest.fn(() => dt);
+
+    beforeEach(() => {
+        writer = new SkillsWriter(fileContext);
+    });
+
+    test('createMetadataCard() should return metadata card', () => {
+        const status = {
+            code: 'skill_invoked_status',
+            message: 'invoked'
+        };
+        const entries = [{ text: 'test1', type: 'text' }, { text: 'test2', type: 'text' }];
+        const duration = 100;
+        const expectedMetadataCard = {
+            created_at: mockDate,
+            type: 'skill_card',
+            skill: {
+                type: 'service',
+                id: skillId
+            },
+            skill_card_type: 'keyword',
+            skill_card_title: {
+                code: 'skills_topics',
+                message: 'Topics'
+            },
+            invocation: {
+                type: 'skill_invocation',
+                id: boxRequestId
+            },
+            status,
+            entries,
+            duration
+        };
+        const metadataCard = writer.createMetadataCard('keyword', 'Topics', status, entries, duration);
+
+        expect(expectedMetadataCard).toEqual(metadataCard);
```

Comment:
Minor: should the two cards be switched, since `metadataCard` is the one being tested?

```suggestion
        expect(metadataCard).toEqual(expectedMetadataCard);
```